lib/tasks/extraneous_dependencies.rake

```diff
@@ -0,0 +1,101 @@
+require "tasks/helpers/compact_index_tasks_helper"
+
+namespace :extraneous_dependencies do
+  def fetch_spec_deps(full_name)
+    spec_uri = URI("https://rubygems.org/quick/Marshal.4.8/#{full_name}.gemspec.rz")
+    http = Net::HTTP.new(spec_uri.host, spec_uri.port)
+    http.use_ssl = true
+
+    request = Net::HTTP::Get.new(spec_uri.request_uri)
+    res = http.request(request)
+
+    raise StandardError, "fetch deps request for #{full_name} failed #{res.inspect}" unless res.code == "200"
+
+    spec_obj = Marshal.load(Gem::Util.inflate(res.body))
+
+    spec_run_deps = spec_obj.dependencies.map do |s|
+      s.name.downcase if s.type == :runtime && Rubygem.where(name: s.name).present?
+    end.compact.sort
+
+    spec_dev_deps = spec_obj.dependencies.map do |s|
+      s.name.downcase if s.type == :development && Rubygem.where(name: s.name).present?
+    end.compact.sort
+
+    [spec_run_deps, spec_dev_deps]
+  end
+
+  desc "Remove dependencies from DB where it doesn't match the gemspec"
+  task clean: :environment do |task|
+    ActiveRecord::Base.logger.level = 1 if Rails.env.development?
```

Comment:
I figured out why I was not seeing any output: except for the counter, which is printed directly to STDOUT, everything is logged to `log/development.log` by default.

Not sure if this change was related to that? The default log level in rails is `:debug` (0), so I guess the task was too verbose by default?

Reply:
Idea of using rails logger is that logs will be sent to datadog this way. if I need the live trail I can use pod logs with grep for task name.

however, in development env this meant I was seeing SQL queries in log/development.log as well. so yes, added this just to remove SQL queries in dev.

Reply:
That makes sense, thanks for explaining!